akka-http-core/src/main/scala/akka/http/scaladsl/model/HttpEntity.scala

```diff
@@ -213,6 +236,9 @@ object HttpEntity {
     def withContentType(contentType: ContentType): Default =
       if (contentType == this.contentType) this else copy(contentType = contentType)
 
+    def withSizeLimit(maxBytes: Long): Default =
+      copy(data = data withAttributes Attributes(SizeLimit(maxBytes, Some(contentLength))))
```

Comment:
Yes, documentation is required.
If you transform the entity in a way that changes the Content-Length and then apply another limit then this new limit will be applied against the new Content-Length, which is the right thing to do IMO.
If you transform the entity in a way that changes the Content-Length and *don't* apply another limit then the *previous* limit (e.g. the configured one) will be applied against the *previous* Content-Length, which is also ok I think.
So, all in all I think this feature is actually behaving nicely overall.